client/src/components/Main/Carousel/Carousel.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+import styled from '@emotion/styled';
+import '@/static/assets/img/bannerImage1.gif';
+import '@/static/assets/img/bannerImage2.gif';
+import '@/static/assets/img/bannerImage3.gif';
+import { greyButton } from '@/static/style/common';
+
+/**
+ * TODO:
+ * 이미지 클릭 시 해당 상품으로 넘어가도록 추후 Link에 product id 달 것
+ */
+const Carousel = () => {
+  const [index, setIndex] = useState(1);
+  const [clickEventActive, setClickEventActive] = useState(false);
+
+  /**
+   * button clickEvent가 일어나지 않았을때
+   * index를 이동합니다. ( 이미지를 보여주기 위해서 )
+   */
+  !clickEventActive &&
+    setTimeout(() => {
+      const standardIndex = index + 1 > 3 ? 1 : index + 1;
+      setIndex(standardIndex);
+    }, 3000);
```

Comment:
```suggestion
useEffect(() => {
    console.log('usee');
    if (!clickEventActive) {
      setTimeout(() => {
        const standardIndex = index + 1 > 3 ? 1 : index + 1;
        setIndex(standardIndex);
      }, 2000);
    }
  }, [index, clickEventActive]);
```
`useEffect`를 써서 index와 clickEventActive에만 반응하도록 하면 좋을거 같은데 어떠신지요

Reply:
위에 state가 index와 clickEventActive뿐이라 useEffect 쓰나안쓰나 같겠다 싶어서 했는데 저도 저런식으로 처음해봐서 다른게 있는지 궁금합니다!

Reply:
저도 다른 state가 없어서 문제는 없다고 생각합니다. 

다만 나중에 state가 생기면 그때는 useEffect를 사용해줘야 하니 염려 차원에서 말씀드렸습니다~!